Assembler.Base/Releasing/PositionCountValidators/BaseAssemblingPositionCountValidator.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.RawAssemblingEntities;
+using Microsoft.Extensions.Logging;
+
+namespace Assembler.Base.Releasing.PositionCountValidators
+{
+    public abstract class BaseAssemblingPositionCountValidator : IValidator<RawMessageInAssembly>
+    {
+        private readonly int _minimumFramesCount;
+        private readonly ILogger _logger;
+
+        protected BaseAssemblingPositionCountValidator(int minimumFramesCount, ILoggerFactory loggerFactory)
+        {
+            _minimumFramesCount = minimumFramesCount;
+            _logger = loggerFactory.CreateLogger(ToString());
+        }
+
+        public bool IsValid(RawMessageInAssembly message)
+        {
+            var framesList = GetFramesList(message).ToArray();
+
+            if (framesList.Length >= _minimumFramesCount)
+            {
+                _logger.LogDebug($"The message [{message.Guid}] passed the validator");
+                return true;
+            }
+
+            _logger.LogDebug($"The message [{message.Guid}] failed the validator");
```

Comment:
I dont think the validator should log, its not his responsibility. The ConditionalReleaser should, and only in case of filtering (as it does)

Reply:
But that way you know the specific one who fails.
I think it's better this way.